app/components/barcode_component.rb

```diff
@@ -3,36 +3,51 @@
 require 'barby/outputter/html_outputter'
 
 class BarcodeComponent < BaseComponent
-  attr_reader :barcode_data, :label, :label_formatter, :tag_options
+  attr_reader :barcode_data, :label, :label_formatter, :barcode_image_url_method, :tag_options
 
-  def initialize(barcode_data:, label:, label_formatter: nil, **tag_options)
+  def initialize(
+    barcode_data:,
+    label:,
+    label_formatter: nil,
+    barcode_image_url_method: nil,
+    **tag_options
+  )
     @barcode_data = barcode_data
     @label = label
     @label_formatter = label_formatter
+    @barcode_image_url_method = barcode_image_url_method
     @tag_options = tag_options
   end
 
+  def barcode_image_url
+    if barcode_image_url_method.respond_to?(:call)
+      barcode_image_url_method.call
+    else
+      barcode_data_url
+    end
+  end
+
   def formatted_data
     formatted_data = barcode_data
     formatted_data = label_formatter.call(formatted_data) if label_formatter
     formatted_data
   end
 
-  def barcode_html
-    html = Barby::Code128.new(barcode_data).to_html(class_name: 'barcode')
-    # The Barby gem doesn't provide much control over rendered output, so we need to manually slice
-    # in accessibility features (label as substitute to illegible inner content).
-    html.gsub(
-      '><tbody>',
-      %( aria-label="#{t('components.barcode.table_label')}"><tbody aria-hidden="true">),
-    )
-  end
-
   def barcode_caption_id
     "barcode-caption-#{unique_id}"
   end
 
   def css_class
-    [*tag_options[:class], 'display-inline-block margin-0']
+    [*tag_options[:class], 'barcode display-inline-block margin-0']
```

Comment:
What is the `barcode` class here?

Reply:
> What is the `barcode` class here?

This is lingering from one of my many earlier attempts, I can clean it up.

Reply:
Removed in 26af3cead.